### PR TITLE
PR 06 User can view kindergartens and link kindergartens to user account

### PR DESCRIPTION
**Pull request by ghost**
```
As a User,
In order to be able link kindergartens to my account,
I would like to see a list of kindergartens and be able to add the ones I am interested in.
```
Goals:
1. Generate kindergarten model with appropriate attributes.  Unit test for model.
2. Create association between user model and kindergarten model so that any user can have any number of chosen kindergartens associated with their network (has_and_belongs_to_many on both).  Create migrations for kindergarten table and link-table.  Unit test for associations.
3. Generate controller and method to display list of all kindergartens for user to choose associations.
4. Generate controller (maybe new) and method(s) to give user buttons to link chosen kindergartens to user account and show linked kindergartens to user in a second list.

[link to PT item](https://www.pivotaltracker.com/story/show/167340584)

### CODE REVIEW

File: db/migrate/20190717180516_create_join_table_user_kindergarten.rb [hunk 1]
```diff
@@ -0,0 +1,8 @@
+class CreateJoinTableUserKindergarten < ActiveRecord::Migration[5.2]
```
Comment (oliverochman) [inline db/migrate/20190717180516_create_join_table_user_kindergarten.rb]: can u delete this migration file if you are not using it

File: features/step_definitions/basic_steps.rb [hunk 2]
```diff
@@ -2,10 +2,32 @@
   visit root_path
 end
 
+Given("the following kindergartens exist") do |table|
+  table.hashes.each do |kindergarten|
+    Kindergarten.create!(kindergarten)
```
Comment (oliverochman) [inline features/step_definitions/basic_steps.rb]: please use factory bot here

**Review by oliverochman — CHANGES_REQUESTED** [review]

File: db/migrate/20190717180516_create_join_table_user_kindergarten.rb (hunk 1 above)
Comment (ghost) [reply db/migrate/20190717180516_create_join_table_user_kindergarten.rb]: We have connected users to kindergartens and kindergartens to users in both directions with a has_and_belongs_to_many relationship.  Then, at the end of the feature test, we link a user to at least one kindergarten so I believe it stores the link information in the Join Table.

File: features/step_definitions/basic_steps.rb (hunk 2 above)
Comment (ghost) [reply features/step_definitions/basic_steps.rb]: Fixed.

File: db/migrate/20190717180516_create_join_table_user_kindergarten.rb (hunk 1 above)
Comment (oliverochman) [reply db/migrate/20190717180516_create_join_table_user_kindergarten.rb]: but you have commented out code in this migration file?

Comment (ghost) [reply db/migrate/20190717180516_create_join_table_user_kindergarten.rb]: The migration file has been removed.

**Review by oliverochman — APPROVED** [review]

**Pull request merged**